assembly.py

```diff
@@ -338,6 +339,27 @@ def parser_assemble_trinity(parser=argparse.ArgumentParser()):
 
 __commands__.append(('assemble_trinity', parser_assemble_trinity))
 
+def gapfill_gap2seq(in_scaffold, in_bam, out_scaffold, gap2seq_opts='', threads=1, mem_limit_gb=4):
+    ''' This step runs the Gap2Seq tool to close gaps between contigs in a scaffold.
+    '''
+    tools.gap2seq.Gap2SeqTool().gapfill(in_scaffold, in_bam, out_scaffold, gap2seq_opts=gap2seq_opts, threads=threads,
+                                        mem_limit_gb=mem_limit_gb)
+
+def parser_gapfill_gap2seq(parser=argparse.ArgumentParser(description='Close gaps between contigs in a scaffold')):
+    parser.add_argument('inScaffold', help='Scaffold with gaps (FASTA with Ns)')
+    parser.add_argument('inBam', help='Input unaligned reads, BAM format.')
+    parser.add_argument('outScaffold', help='Output assembly.')
+    parser.add_argument('--gap2seqOpts', dest='gap2seq_opts', default='', help='(advanced) Extra command-line options to pass to Gap2Seq')
+    parser.add_argument('--threads', default=1, type=int, help='Number of threads (default: %(default)s)')
```

Comment:
- I could see it being useful to expose the `time_limit_minutes` optional parameter. But maybe to reduce confusion (since "limit" implies a hard limit) you could call it something like "soft_time_limit" or similar.
- If we pass `None` to `--threads` or `--memLimitGb`, can Gap2Seq do something smart like pick a limit on its own? Or is it always mandatory for the user to specify these numbers?

Reply:
For memory Gap2Seq's default is 20GB, which seems high, so I put an argparse default of 4GB.
For threads, passing 0 means use all available cores.